clients/mesh_rpc_client/src/ws_client.ts

```diff
@@ -0,0 +1,237 @@
+        let snapshotID = ''; // New snapshot
+        let ordersInfosLen = 1;
+
+        let i = 0;
+        let rawAcceptedOrderInfos: RawAcceptedOrderInfo[] = [];
+        while (ordersInfosLen === perPage) {
+            const page = i;
+            const getOrdersResponse: GetOrdersResponse = await (this._wsProvider as any).send('mesh_getOrders', [
+                page,
+                perPage,
+                snapshotID,
+            ]);
+            snapshotID = getOrdersResponse.snapshotID;
+            rawAcceptedOrderInfos = [...rawAcceptedOrderInfos, ...getOrdersResponse.ordersInfos];
+            ordersInfosLen = getOrdersResponse.ordersInfos.length;
+            i++;
+        }
+        const allOrdersInfos = WSClient._convertRawAcceptedOrders(rawAcceptedOrderInfos);
+        return allOrdersInfos;
+    }
+    /**
+     * Subscribe to the 'orders' topic and receive order events from Mesh. This method returns a
+     * subscriptionId that can be used to `unsubscribe()` from this subscription.
+     * @param   cb   callback function where you'd like to get notified about order events
+     * @return subscriptionId
+     */
+    public async subscribeToOrdersAsync(cb: (orderEvents: OrderEvent[]) => void): Promise<string> {
+        assert.isFunction('cb', cb);
+        const orderEventsSubscriptionId = await this._wsProvider.subscribe('mesh_subscribe', 'orders', []);
+        const id = uuid();
+        this._subscriptionIdToMeshSpecificId[id] = orderEventsSubscriptionId;
+
+        const orderEventsCallback = (eventPayload: OrderEventPayload) => {
+            this._subscriptionIdToMeshSpecificId[id] = eventPayload.subscription;
+            const rawOrderEvents: RawOrderEvent[] = eventPayload.result;
+            const orderEvents: OrderEvent[] = [];
+            rawOrderEvents.forEach(rawOrderEvent => {
+                const orderEvent = {
+                    orderHash: rawOrderEvent.orderHash,
+                    signedOrder: orderParsingUtils.convertOrderStringFieldsToBigNumber(rawOrderEvent.signedOrder),
+                    kind: rawOrderEvent.kind,
+                    fillableTakerAssetAmount: new BigNumber(rawOrderEvent.fillableTakerAssetAmount),
+                    txHashes: rawOrderEvent.txHashes,
+                };
+                orderEvents.push(orderEvent);
+            });
+            cb(orderEvents);
+        };
+        this._wsProvider.on(orderEventsSubscriptionId, orderEventsCallback as any);
+        return id;
+    }
+    /**
+     * Subscribe to the 'heartbeat' topic and receive an ack from the Mesh every 5 seconds. This method
+     * returns a subscriptionId that can be used to `unsubscribe()` from this subscription.
+     * @param   cb   callback function where you'd like to get notified about heartbeats
+     * @return subscriptionId
+     */
+    public async subscribeToHeartbeatAsync(cb: (ack: string) => void): Promise<string> {
```

Comment:
Do we need to expose this as a public method if we are also subscribing to the heartbeat endpoint automatically under the hood? It seems like we should do one or the other.

Reply:
Agreed. Made it private.